File: EtudeBackend/EtudeBackend/Features/TrainingRequests/Entities/Course.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;
using EtudeBackend.Features.TrainingRequests.DTOs;
using EtudeBackend.Shared.Data;

namespace EtudeBackend.Features.TrainingRequests.Entities;

public class Course
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public string Name { get; set; } = string.Empty;

    [Column(TypeName = "text")]
    public string Description { get; set; } = string.Empty;

    public CourseType Type { get; set; } = CourseType.NotImplemented;

    public CourseTrack Track { get; set; } = CourseTrack.NotImplemented;

    public CourseFormat Format { get; set; } = CourseFormat.NotImplemented;

    public string TrainingCenter { get; set; } = string.Empty;

    public string Link { get; set; } = string.Empty;

    public Guid EmployeeId { get; set; }

    public DateOnly StartDate { get; set; }

    public DateOnly EndDate { get; set; }

    public string Price { get; set; }

    [Column(TypeName = "text")]
    public string EducationGoal { get; set; } = string.Empty;

    public bool IsActive { get; set; } = false;

    [Column(TypeName = "timestamp with time zone")]
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;

    [Column(TypeName = "timestamp with time zone")]
    public DateTimeOffset? UpdatedAt { get; set; }

    public ApplicationUser Learner { get; set; }

    public virtual ICollection<Application> Applications { get; set; } = new List<Application>();
    public virtual ICollection<UserStatistics> Statistics { get; set; } = new List<UserStatistics>();
}

public enum CourseType
{
    [EnumMember(Value = "Course")]
    Course, // Курс

    [EnumMember(Value = "Conference")]
    Conference, // Конференция

    [EnumMember(Value = "Certification")]
    Certification, // Сертификация

    [EnumMember(Value = "Workshop")]
    Workshop, // Мастер-класс

    NotImplemented
}

public enum CourseTrack
{
    [EnumMember(Value = "Soft Skills")]
    SoftSkills,

    [EnumMember(Value = "Hard Skills")]
    HardSkills,

    [EnumMember(Value = "Management Skills")]
    ManagementSkills,

    NotImplemented
}

public enum CourseFormat
{
    [EnumMember(Value = "Online")]
    Online,

    [EnumMember(Value = "Offline")]
    Offline,
    NotImplemented
}

File: EtudeWeb/src/shared/types/course.ts
/**
 * Типы форматов курса
 */
export enum CourseFormat {
  /**
   * Очное обучение
   */
  Offline = 'Offline',

  /**
   * Онлайн обучение
   */
  Online = 'Online'
}

/**
 * Типы курсов
 */
export enum CourseType {
  /**
   * Курс
   */
  Course = 'Course',

  /**
   * Конференция
   */
  Conference = 'Conference',

  /**
   * Сертификация
   */
  Certification = 'Certification',

  /**
   * Мастер-класс
   */
  Workshop = 'Workshop'
}

/**
 * Направления курсов
 */
export enum CourseTrack {
  /**
   * Профессиональные навыки
   */
  HardSkills = 'HardSkills',

  /**
   * Гибкие навыки
   */
  SoftSkills = 'SoftSkills',

  /**
   * Управленческие навыки
   */
  ManagementSkills = 'ManagementSkills'
}

/**
 * Интерфейс для обучающегося
 */
export interface CourseLearner {
  /**
   * Идентификатор пользователя
   */
  id: string

  /**
   * Имя пользователя
   */
  name: string

  /**
   * Фамилия пользователя
   */
  surname: string

  /**
   * Отчество пользователя
   */
  patronymic?: string

  /**
   * Должность пользователя
   */
  position?: string

  /**
   * Подразделение пользователя
   */
  department?: string

  /**
   * Является ли руководителем
   */
  isLeader?: boolean
}

/**
 * Интерфейс для курса
 */
export interface Course {
  /**
   * Идентификатор курса
   */
  course_id: string

  /**
   * Название курса
   */
  course_name: string

  /**
   * Описание курса
   */
  course_description?: string

  /**
   * Тип курса
   */
  course_type: CourseType

  /**
   * Направление курса
   */
  course_track: CourseTrack

  /**
   * Формат курса
   */
  course_format: CourseFormat

  /**
   * Учебный центр
   */
  course_trainingCenter?: string

  /**
   * Дата начала курса
   */
  course_startDate: string

  /**
   * Дата окончания курса
   */
  course_endDate: string

  /**
   * Ссылка на курс или место проведения
   */
  course_link?: string

  /**
   * Цена курса
   */
  course_price?: string | number

  /**
   * Цель обучения
   */
  course_educationGoal?: string

  /**
   * Обучающийся
   */
  course_learner?: CourseLearner
}
